hudi-io/src/main/java/org/apache/hudi/storage/HoodieLocation.java

```diff
@@ -186,6 +186,51 @@ public URI toUri() {
     return uri;
   }
 
+  /**
+   * Returns a qualified location object.
+   *
+   * @param defaultUri if this location is missing the scheme or authority
+   *                   components, borrow them from this URI.
+   * @return this location if it contains a scheme and authority, or
+   * a new path that includes a path and authority and is fully qualified.
+   */
+  @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
+  public HoodieLocation makeQualified(URI defaultUri) {
```

Comment:
this ought to be a static method? I think we should instead make this a new constructor in `HoodieLocation`?